crates/driver/src/domain/competition/solution/scoring.rs

```diff
@@ -85,16 +88,17 @@ impl Trade {
     /// fees have been applied.
     ///
     /// Denominated in SURPLUS token
-    fn surplus(&self) -> Option<eth::Asset> {
+    fn surplus_over_limit_price(
```

Comment:
nit: here as well I think a private struct with buy and sell fields (which can be returned by adjust_quote_to_order_limits) would be nice to avoid messing up sell and buy amount.

In that case I also think we can leave the name concise as `fn surplus(&self, limit: Amounts)` without causing confusion (maybe we should update the funcioin comment)

Reply:
Used `compentition::Amounts` struct